pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -731,16 +731,21 @@ public CompletableFuture<Boolean> isNamespaceBundleOwned(NamespaceBundle bundle)
         return pulsar.getLocalMetadataStore().exists(ServiceUnitUtils.path(bundle));
     }
 
-    public Map<String, NamespaceOwnershipStatus> getOwnedNameSpacesStatus() throws Exception {
-        NamespaceIsolationPolicies nsIsolationPolicies = this.getLocalNamespaceIsolationPolicies();
-        Map<String, NamespaceOwnershipStatus> ownedNsStatus = new HashMap<String, NamespaceOwnershipStatus>();
-        for (OwnedBundle nsObj : this.ownershipCache.getOwnedBundles().values()) {
-            NamespaceOwnershipStatus nsStatus = this.getNamespaceOwnershipStatus(nsObj,
-                    nsIsolationPolicies.getPolicyByNamespace(nsObj.getNamespaceBundle().getNamespaceObject()));
-            ownedNsStatus.put(nsObj.getNamespaceBundle().toString(), nsStatus);
-        }
-
-        return ownedNsStatus;
+    public CompletableFuture<Map<String, NamespaceOwnershipStatus>> getOwnedNameSpacesStatusAsync() {
+       return getLocalNamespaceIsolationPoliciesAsync()
+                .thenCompose(namespaceIsolationPolicies -> {
+                    List<CompletableFuture<OwnedBundle>> ownedBundleFutures = Collections.unmodifiableList(
+                            new ArrayList<>(ownershipCache.getOwnedBundlesAsync().values()));
+                    return FutureUtil.waitForAll(ownedBundleFutures)
```

Comment:
You create new list -> `new ArrayList<>`, so no need to use `unmodifiableList`

Reply:
Since the `ownedBundleFutures ` needs to be used elsewhere, this method will guarantee that it will not be used incorrectly to generate an error later.

Reply:
`ownedBundleFutures` is an internal field.

Reply:
fixed

Reply:
fixed